### PR TITLE
bgp: remove metallb-bgp documentation

### PR DESCRIPTION
**Pull request by harsimran-pabla**
Removing metallb-bgp integration documentation as well as adding notes to upgrade to 1.17 document with references to removal of metallb-bgp.

Related to: https://github.com/cilium/cilium/pull/36191

### CODE REVIEW

**Review by joestringer — APPROVED** [review]
:tada:

**Review by rastislavs — APPROVED** [review]

File: Documentation/operations/upgrade.rst [hunk 1]
```diff
@@ -343,6 +344,9 @@ Removed Options
 * The previously deprecated built-in WireGuard userspace-mode fallback (Helm ``wireguard.userspaceFallback``)
   has been removed. Users of WireGuard transparent encryption are required to use a Linux kernel with
   WireGuard support.
+* The previously deprecated ``metallb-bgp`` flags ``bgp-config-path``, ``bgp-announce-lb-ip``
+  and ``bgp-announce-pod-cidr`` has been removed. Users are required to use Cilium BGP
+  control plane for BGP advertisements.
```
Comment (learnitall) [inline Documentation/operations/upgrade.rst]:
```suggestion
* The deprecated ``metallb-bgp`` flags ``bgp-config-path``, ``bgp-announce-lb-ip``
  and ``bgp-announce-pod-cidr`` have been removed. Users are now required to use Cilium BGP
  control plane for BGP advertisements.
```

File: Documentation/operations/upgrade.rst [hunk 2]
```diff
@@ -364,6 +368,9 @@ Helm Options
   to validate certificates with expirations longer than 825 days.
 * The Helm option ``hubble.relay.dialTimeout`` has been deprecated (now a no-op)
   and will be removed in Cilium 1.18.
+* The ``metallb-bgp`` integration Helm options ``bgp.enabled``, ``bgp.announce.podCIDR``, and
+  ``bgp.announce.loadbalancerIP`` has been removed. Users are required to use Cilium BGP
```
Comment (learnitall) [inline Documentation/operations/upgrade.rst]:
```suggestion
  ``bgp.announce.loadbalancerIP`` have been removed. Users are now required to use Cilium BGP
```

**Review by learnitall — CHANGES_REQUESTED** [review]
Couple of small typos and grammar suggestions, but LGTM! Thank you!

**Comment by harsimran-pabla** [comment]
Thanks @learnitall, PR updated with requested changes.

**Review by learnitall — APPROVED** [review]
Thank you! :rocket:

**Pull request merged**